sdk/core/azure-core/azure/core/_pipeline_client.py

```diff
@@ -170,3 +184,37 @@ def _build_pipeline(self, config, **kwargs): # pylint: disable=no-self-use
             transport = RequestsTransport(**kwargs)
 
         return Pipeline(transport, policies)
+
+
+    def send_request(self, request, **kwargs):
+        # type: (HTTPRequestType, Any) -> HTTPResponseType
+        """**Provisional** method that runs the network request through the client's chained policies.
+
+        This method is marked as **provisional**, meaning it may be changed in a future release.
+
+        >>> from azure.core.rest import HttpRequest
+        >>> request = HttpRequest('GET', 'http://www.example.com')
+        <HttpRequest [GET], url: 'http://www.example.com'>
+        >>> response = client.send_request(request)
+        <HttpResponse: 200 OK>
+
+        :param request: The network request you want to make. Required.
+        :type request: ~azure.core.rest.HttpRequest
+        :keyword bool stream: Whether the response payload will be streamed. Defaults to False.
+        :return: The response of your network call. Does not do error handling on your response.
+        :rtype: ~azure.core.rest.HttpResponse
+        # """
+        rest_request, request_to_run = _prepare_request(request)
+        return_pipeline_response = kwargs.pop("_return_pipeline_response", False)
+        pipeline_response = self._pipeline.run(request_to_run, **kwargs)  # pylint: disable=protected-access
+        response = pipeline_response.http_response
+        if rest_request:
+            response = _to_rest_response(response)
+            if not kwargs.get("stream", False):
+                response.read()
```

Comment:
Will we leak a stream if `read` fails?

Reply:
good point, I'm going to add a try except when I read the response, and close the response before throwing the error.  thanks!